src/kotlin_editor_export_plugin.cpp

```diff
@@ -6,30 +6,37 @@
 void KotlinEditorExportPlugin::_export_begin(const Set<String>& p_features, bool p_debug, const String& p_path,
                                              int p_flags) {
     // Add mandatory jars to pck
-    String main_jar{"res://build/libs/main.jar"};
-    String bootstrap_jar{"res://build/libs/godot-bootstrap.jar"};
+    String main_jar;
+    String bootstrap_jar;
+    if (p_path.ends_with(".apk")) {
+        main_jar = "res://build/libs/main-dex.jar";
+        bootstrap_jar = "res://build/libs/godot-bootstrap-dex.jar";
+    } else {
+        main_jar = "res://build/libs/main.jar";
+        bootstrap_jar = "res://build/libs/godot-bootstrap.jar";
+    }
     add_file(main_jar, FileAccess::get_file_as_array(main_jar), false);
     add_file(bootstrap_jar, FileAccess::get_file_as_array(bootstrap_jar), false);
 
     // Copy JRE for desktop platforms
-#if defined _WIN32 || defined _WIN64 || defined __linux__ || defined __APPLE__
-    const Vector<String>& path_split = p_path.split("/");
-    String export_dir{p_path.replace(path_split[path_split.size() - 1], "")};
-    Error error;
-    DirAccess* dir_access{DirAccess::open(export_dir, &error)};
-    if (error == OK) {
-        if (dir_access->copy_dir(
-                ProjectSettings::get_singleton()->globalize_path("res://jre"),
-                vformat("%s/jre", dir_access->get_current_dir())
-        ) != OK) {
-            LOG_ERROR("Cannot copy jre folder to export folder, please make sure you created a jre in project "
-                      "root folder using jlink.")
+    if (!p_path.ends_with(".apk")) {

```

Comment:
This will not hold up anymore with 3.2.4!
As 3.2.4 will add bundle support. Not sure if just checking for `aar` as well is enough but most probably it is

Reply:
https://github.com/godotengine/godot/pull/42185
and https://github.com/godotengine/godot-proposals/issues/1111 respecively